### PR TITLE
fix: npe in join utils

### PR DESCRIPTION
**Pull request by bryans99**
Updates join utilities to ensure arrays/objects exist before using them.

### CODE REVIEW

File: src/utils/LookmlDiagrammer/join-utils.ts [hunk 1]
```diff
@@ -41,14 +41,14 @@ export function getPkJoinPathObj(
   pkTableRef: DiagramField[],
   pkFieldIndex: number
 ) {
-  return {
+  return ({
     viewName: join.name,
     fieldIndex: pkFieldIndex,
-    selector: pkTableRef[pkFieldIndex].name.replace('.', '-'),
+    selector: pkTableRef[pkFieldIndex]?.name?.replace('.', '-'),
```
Comment (jeremytchang) [inline src/utils/LookmlDiagrammer/join-utils.ts:47]: Do we have a good default value for `selector`? Not sure how `selector` is referenced later, but could leave it undefined, which could still lead to issues later. Looks like you set a default value on all the other values to be safe.

File: src/utils/LookmlDiagrammer/join-utils.ts [hunk 2]
```diff
@@ -64,14 +64,14 @@ export function getBaseJoinPathObj(
   baseTableRef: DiagramField[],
   fieldIndex: number
 ) {
-  return {
+  return ({
     viewName: baseTableId,
     fieldIndex: fieldIndex,
-    selector: baseTableRef[fieldIndex].name.replace('.', '-'),
+    selector: baseTableRef[fieldIndex]?.name?.replace('.', '-'),
```
Comment (jeremytchang) [inline src/utils/LookmlDiagrammer/join-utils.ts:70]: Same as my previous comment

File: src/utils/LookmlDiagrammer/join-utils.ts [hunk 3]
```diff
@@ -41,14 +41,14 @@ export function getPkJoinPathObj(
...
+  return ({
     viewName: join.name,
     fieldIndex: pkFieldIndex,
-    selector: pkTableRef[pkFieldIndex].name.replace('.', '-'),
+    selector: pkTableRef[pkFieldIndex]?.name?.replace('.', '-'),
     type: 'core',
     joinName: join.name,
     joinObj: join,
-  }
+  } as unknown) as DiagramField
```
Comment (jeremytchang) [inline src/utils/LookmlDiagrammer/join-utils.ts:51]: Guessing this came from internal testing?

File: src/utils/LookmlDiagrammer/join-utils.ts (hunk 1 above)
Comment (bryans99) [reply src/utils/LookmlDiagrammer/join-utils.ts:47]: Unfortunately not. I don't have a repro either. All I'm trying to do is fix the NPE. What happens after this I do not know

File: src/utils/LookmlDiagrammer/join-utils.ts (hunk 2 above)
Comment (bryans99) [reply src/utils/LookmlDiagrammer/join-utils.ts:70]: See previous comment

File: src/utils/LookmlDiagrammer/join-utils.ts (hunk 3 above)
Comment (jkaster) [reply src/utils/LookmlDiagrammer/join-utils.ts:51]: the PR is in response to a customer ticket, if that's what you mean?

**Review by jkaster — APPROVED** [review]
Looks reasonable. Without clear repro steps it's probably worth getting this published. We can always revert if it makes the situation worse, but that seems unlikely.

File: src/utils/LookmlDiagrammer/join-utils.ts (hunk 3 above)
Comment (jeremytchang) [reply src/utils/LookmlDiagrammer/join-utils.ts:51]: Ah i meant this specific line's type casting.

**Review by Itzaprado — COMMENTED** [review]
LGTM, I tested if any errors arise with the changes and no, it works fine. Once the customer confirms that it solves their problem, we could merge it.

**Review by Itzaprado — APPROVED** [review]

**Pull request merged**